### PR TITLE
Config, Config, and Config...

### PR DESCRIPTION
**Pull request by dayvson**
- Update config file (To enable database driver selection)
- Refactoring on database package
- Passing config object instead of config path
- Refactoring tests

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "SWAP_DIRECTORY": "/tmp/",
   "LOGFILE": "/tmp/output.log",
   "PORT": "8000",
+  "DBDRIVER": "memory",
```
Comment (flavioribeiro) [inline config.json]: what if we just check if `MONGODB_HOST != ""` to select the right driver?

Comment (flavioribeiro) [inline config.json]: if it's different we use mongo, otherwise we use memory

File: db/db.go [hunk 2]
```diff
@@ -21,3 +24,11 @@ type Storage interface {
 
 	ClearDatabase() error
 }
+
+func GetDatabase(config gonfig.Gonfig) (Storage, error) {
+	driver, _ := config.GetString("DBDRIVER", "memory")
+	if driver == "mongo" {
+		return getMongoDatabase(config)
+	}
+	return getMemoryDatabase()
```
Comment (flavioribeiro) [inline db/db.go]: this will close #119

File: db/mongo.go [hunk 3]
```diff
@@ -1,43 +1,46 @@
...
-	instance.db = session.DB("snickers")
-	return instance, nil
+func getMongoDatabase(cfg gonfig.Gonfig) (Storage, error) {
+	var initErr error
+	databaseMongoInit.Do(func() {
+		mongoInstance = &mongoDatabase{}
+		mongoHost, _ := cfg.GetString("MONGODB_HOST", "")
+		session, err := mgo.Dial(mongoHost)
+		if err != nil {
+			initErr = err
```
Comment (flavioribeiro) [inline db/mongo.go]: can you write a test for the case when mongo fails? like trying to connect to a random IP.. just to cover this `if`

**Comment by flavioribeiro** [comment]
That's a really great pull request, @dayvson. Lots of improvement/refactory. Thank you very much for this! 🎆

File: config.json (hunk 1 above)
Comment (dayvson) [inline config.json]: I don't think so, because I want in the future to add support to Redis or another database.

**Comment by flavioribeiro** [comment]
🎆 🎆 🎆

**Pull request merged**